src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (type.getSimpleName().equals("Truck")) {
```

Comment:
I would suggest more compact expression without hardcode
```suggestion
        if (type == Bulldozer.class) {
```

Reply:
its won't work without simple name couse we comparing string with type / i tried , but it didn't work(

Reply:
I agree with @AshaJaneBrown, better compare type Class than String